### PR TITLE
rule-reload: enable rule-reload for -s and -S run as well

### PR DESCRIPTION
**Pull request by norg**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

https://redmine.openinfosecfoundation.org/issues/2848

Describe changes:
- this enables rule-reload via USR2 for -s and -S runs

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

- PR norg-pcap: https://buildbot.openinfosecfoundation.org/builders/norg-pcap/builds/71
- PR norg: https://buildbot.openinfosecfoundation.org/builders/norg/builds/73

### CODE REVIEW

File: src/suricata.c [hunk 1]
```diff
@@ -2860,28 +2859,16 @@ static void SuricataMainLoop(SCInstance *suri)
         }
 
         if (sigusr2_count > 0) {
```
Comment (jlucovsky) [inline src/suricata.c]: Suggest `if (sigusr2_count)` with `unlikely` here (and for the other conditionals in this loop).

Comment (victorjulien) [reply src/suricata.c]: Not sure if this is important, as this is a loop that sleeps a lot. Btw this is also not part of the changeset that is being proposed, so a 'optimization' ticket would be the proper place normally.

Comment (jlucovsky) [reply src/suricata.c]: Given that, then I agree ... using `unlikely, likely` are not necessary.

**Comment by victorjulien** [comment]
Merged in #4108, thanks Andreas!

**Pull request merged**